File: problems/template2.py
# template file
import json
from typing import List


class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


def createLinkedList(nums):
    head = None
    for num in reversed(nums):
        newNode = ListNode(num, head)
        head = newNode
    
    return head


def linkedListToVector(node):
    intList = []
    while (node != None):
        intList.append(node.val)
        node = node.next
    
    return intList


# insert Solution class here


with open("../testcase-2-add-two-numbers.json", "r") as read_file:
    data = json.load(read_file)

sol = Solution()
isOk = True

for i in range(len(data["input"]["l1"])):
    listL1 = data["input"]["l1"]
    listL2 = data["input"]["l2"]

    for i, nums in enumerate(listL1):
        l1 = createLinkedList(listL1[i])
        l2 = createLinkedList(listL2[i])

        print("l1: ", l1.val)
        print("l2: ", l2.val)

        solutionNode = sol.addTwoNumbers(l1, l2)
        solutionList = linkedListToVector(solutionNode)
        for j, solution in enumerate(solutionList):
            if solution != data["expected"][j]:
                isOk = False;
                break;

if isOk:
    # write json to file
    output = {
        "result": "OK"
    }
    json_object = json.dumps(output, indent=4)
    print(json_object)

print("test completed")

File: problems/template2.js
// template file
const tc = require("../testcase-2-add-two-numbers.json")

class ListNode {
    constructor(val, next) {
        this.val = (val===undefined ? 0 : val)
        this.next = (next===undefined ? null : next)
    }
}

function createLinkedList(nums) {
    let head = null
    nums.forEach(num => {
        let newNode = new ListNode(num, head)
        head = newNode
    });

    return head
}

function linkedListToVector(node) {
    let intList = []
    while (node != null) {
        intList.push(node.val)
        node = node.next
    }

    return intList
}

// insert Solution class here


let isOk = true
const listL1 = tc.input.l1
const listL2 = tc.input.l2
for (let i = 0; i < listL1.length; i++) {
    const l1 = createLinkedList(listL1[i])
    const l2 = createLinkedList(listL2[i])

    const solutionNode = addTwoNumbers(l1, l2)
    const solutionList = linkedListToVector(solutionNode)

    for (let j = 0; j < solutionList.length; j++) {
        if (solutionList[j] != tc.expected[i][j]) {
            isOk = false
            break
        }   
    }
}

if (isOk) {
    const output = {
        "result": "OK"
    }
    const data = JSON.stringify(output)
    console.log(data)
}

console.log("test completed")